### DISCUSSION TITLE
Make "add url" function works with BibLaTeX format.

### DISCUSSION THREAD

**Discussion by earthmoon** [retorquere/zotero-better-bibtex#2513] [Ideas]
Hello,

There is a function of [Adding URLs to BibTeX export](https://retorque.re/zotero-better-bibtex/installation/preferences/export/#add-urls-to-bibtex-export). When export with BibLaTex format, this function won't work, but most users may seem there shouldn't be different setting between BibTex and BibLaTeX, so will it be better to set this function works for BibLaTeX too?

Thanks.

**Comment by retorquere** [comment]
I need a debug log ID.

**Reply by earthmoon** [reply]
> I need a debug log ID.

I don't know why I can't send it in Zotero, but I can upload it here. 
[SXVJEGY9-apse.zip](https://github.com/retorquere/zotero-better-bibtex/files/11484108/SXVJEGY9-apse.zip)

Thanks.

**Reply by retorquere** [reply]
I need you to right click the item in Zotero and send a BBT debug log from that popup menu. The debug log ID will have `-refs-` in it.

**Reply by earthmoon** [reply]
> I need you to right click the item in Zotero and send a BBT debug log from that popup menu. The debug log ID will have `-refs-` in it.

This one: YPGJ6VLZ-apse . Thanks.

**Reply by retorquere** [reply]
No, that doesn't have `-refs-` in the ID, this one is from the help menu. I need one which you get from right-clicking the item in Zotero, from the popup menu you then get.

**Reply by earthmoon** [reply]
> No, that doesn't have `-refs-` in the ID, this one is from the help menu. I need one which you get from right-clicking the item in Zotero, from the popup menu you then get.

This one: RIC5W5QI-refs-apse ? Thanks.

**Comment by earthmoon** [comment]
Okay. When choose the "no" in the [Adding URLs to BibTeX export](https://retorque.re/zotero-better-bibtex/installation/preferences/export/#add-urls-to-bibtex-export), press "Ctrl + Shift + c" will contain URLs and DOI together.

**Reply by retorquere** [reply]
Above you said you were exporting to BibLaTeX format? I'm not sure what you are asking right now.

**Reply by earthmoon** [reply]
> Above you said you were exporting to BibLaTeX format? I'm not sure what you are asking right now.

I selected the "Better BibLaTeX" as the quick copy format. I selected the "no" in the [Adding URLs to BibTeX export](https://retorque.re/zotero-better-bibtex/installation/preferences/export/#add-urls-to-bibtex-export), and pressed "Ctrl + Shift + c", then I found that the clipboard contain URLs and DOI together. 

If I select the "Better BibTeX" as the quick copy format.  "Ctrl + Shift + c" wouldn't contain  URLs and DOI together.

**Reply by retorquere** [reply]
Yeah because these are two different export formats. Better BibLaTeX gives different output that Better BibTeX.

**Reply by earthmoon** [reply]
> ort formats. Better BibLaTeX gives different output that Better BibTe

Okay. I know now.

**Reply by retorquere** [reply]
Sorry, that was supposed to say 

> Better BibLaTeX gives different output that Better BibTeX.

**Reply by earthmoon** [reply]
I have one question I don't understand and I would like to ask you.

Why did you added this funciton of [Adding URLs to BibTeX export](https://retorque.re/zotero-better-bibtex/installation/preferences/export/#add-urls-to-bibtex-export) for BibTeX but not for BibLaTeX?

Thanks.

**Reply by retorquere** [reply]
Because BibTeX by default does not support `url` fields, and BibLaTeX does, so for BibLaTeX I don't need to ask.

**Reply by retorquere** [reply]
> I selected the "no" in the Adding URLs to BibTeX export

That only affects Better BibTeX exports, not Better BibLaTeX.

**Reply by earthmoon** [reply]
> Because BibTeX by default does not support `url` fields, and BibLaTeX does, so for BibLaTeX I don't need to ask.

What will happen if BibTeX file contains **url** field? And another question: You does so good in BibTeX with Zotero, why doesn't you become a member of Zotero developer and make the function of Better BibTeX built in the Zotero?

**Reply by retorquere** [reply]
> What will happen if BibTeX file contains **url** field?

It could break compilation, because the `url` field is a verbatim field, with different parsing rules. Biblatex already knows it's a verbatim field, so it knows how to parse it. You can tell bibtex how to parse it by loading extra packages, but those aren't loaded by default.

> And another question: You does so good in BibTeX with Zotero, why doesn't you become a member of Zotero developer and make the function of Better BibTeX built in the Zotero?

It wouldn't really change things for the better. There's very little I cannot do right now, and me and the Zotero team move at a different pace, it would be a frustrating experience for both sides.

**Reply by earthmoon** [reply]
Okay. I know now. Thanks.
